pkg/networks/usernet/client.go

```diff
@@ -74,11 +76,18 @@ func (c *Client) ResolveAndForwardSSH(ipAddr string, sshPort int) error {
 }
 
 func (c *Client) ResolveIPAddress(ctx context.Context, vmMacAddr string) (string, error) {
-	timeout := time.After(2 * time.Minute)
+	timeout := 2 * time.Minute
+	customTimeout := os.Getenv("CUSTOM_TIMEOUT")
+	if customTimeout != "" {
+		if parsedTimeout, err := strconv.Atoi(customTimeout); err == nil {
+			timeout = time.Duration(parsedTimeout) * time.Minute
+		}
+	}
+	timeoutChan := time.After(timeout)
 	ticker := time.NewTicker(500 * time.Millisecond)
 	for {
 		select {
-		case <-timeout:
+		case <-timeoutChan:
```

Comment:
Maybe it's better to control the timeout with the [context](https://pkg.go.dev/context#WithTimeout)?

```suggestion
		case <-ctx.Done():
```

https://go.dev/blog/context